### ISSUE TITLE
[bug] Trust Wallet icon does not respond on mobile

### ISSUE THREAD

**Issue by redrumlive** [rainbow-me/rainbowkit#1476]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

1.0.8

### wagmi Version

1.3.10

### Current Behavior

The trust wallet icon does not respond in the trust wallet app. Clicking walletconnect or metamask icons works as alternative inside TW.

The issue can also be found on the main build at https://www.rainbowkit.com/

### Expected Behavior

Once clicking on the icon the wallet should connect, same as clicking on connectwallet inside the TW app.

### Steps To Reproduce

Open trust wallet, head to https://www.rainbowkit.com/, connect and select the Trust Wallet icon.

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

https://www.rainbowkit.com/

**Comment by DanielSinclair** [comment]
@redrumlive Am unable to reproduce this issue. Are you seeing any console logs on the mobile webpage? Are you using a VPN or have any unique network characteristics? Is Trust Wallet installed and active? This is either a quiet failure in WalletConnect or with the Trust Wallet mobile deep link on your device.

**Comment by redrumlive** [comment]
Hey @DanielSinclair using a mobile device directly without VPN usage, I've tried Google Pixel and Iphone 13, both produce the same result on different networks. Did you test it on a mobile emulator? The problem only occurs within the TrustWallet app, clicking on TrustWallet icon in a web browser works perfectly. But inside the TW app the icon isn't responding.

I've also asked a few colleagues to test it out for me and all of them get the same result, so a non responding icon directly on the RainbowKit web.

**Comment by DanielSinclair** [comment]
@redrumlive What region are you in? WalletConnect v2's servers are blocked in certain regions, including by the Great Firewall. Am still hoping to see more failover improvements from their team.

**Comment by redrumlive** [comment]
@DanielSinclair We've tried with various devices all around Europe and the US.

**Comment by DanielSinclair** [comment]
I see. There is an issue with the `window.trustwallet` provider in the embedded Trust Wallet browser. I'll reach out to their team to report the issue.

**Comment by DanielSinclair** [comment]
Still seeing this issue after filing a bug report. Awaiting guidance from the Trust Wallet team.

**Comment by yasssuz** [comment]
We are taking a look into it 👍

**Comment by ArtemGoryunov** [comment]
@DanielSinclair 
I think it's issue on rainbokit side.
From TrustWallet side, app injects web3provider and it works with Browser Wallet button.

According to this logic:
```
  const isTrustWalletInjected = Boolean(getTrustWalletInjectedProvider());
  const shouldUseWalletConnect = !isTrustWalletInjected;
```

if no TW web3provider is not injected, rainbowkit should show walletconnect popup. It doesn't show on mobile.

**Comment by DanielSinclair** [comment]
@ArtemGoryunov For Trust Wallet's dApp browser, we rely on the injected provider, not WalletConnect. The team has acknowledged an issue with their `window.trustwallet` provider that we favor for this flow.

**Comment by ArtemGoryunov** [comment]
I can confirm that provider is injected. Any other dApps works

**Comment by DanielSinclair** [comment]
This is resolved in version [1.1.4](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%401.1.4)
